### PR TITLE
Closes #1145: add type assertions and edge-case coverage to MCP ability integration tests

### LINKED ISSUE #1145: MCP ability integration tests only assert key presence — missing type and edge case coverage
> 🤖 AI-generated — created by an automated pipeline. Review before acting on this.

**Context**
The integration tests for `GetMediaStatus`, `GetStats`, `GetNextgenCoverage`, `GetAccount`, and `OptimizeMedia` are structurally identical and only verify that the correct keys are present in the response. No type assertions, value range checks, sensitive field stripping verification, or error-path coverage beyond permission-denied are included.

**Files affected**
- `Tests/Integration/classes/Abilities/GetStats/RegisterAbilityTest.php`
- `Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php`
- `Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php`
- `Tests/Integration/classes/Abilities/GetAccount/RegisterAbilityTest.php`
- `Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php`

**Expected behavior**
Integration tests comprehensively validate MCP ability outputs by asserting value types, verifying sensitive fields are stripped, and covering error paths.

**Acceptance Criteria**
- [ ] Output value types are asserted for numeric fields (`count_optimized`, `original_size`, `savings_percent`, etc.)
- [ ] `api_key` is verified absent from `GetAccount` output
- [ ] Non-existent attachment path case for `GetMediaStatus` is tested (once the existence guard is implemented)
- [ ] Error response structure and code are validated when invalid inputs are provided
- [ ] All five ability tests are updated with consistent, comprehensive coverage

**Development steps**
- [ ] Review current test structure in all five ability test files
- [ ] Define comprehensive assertion strategy (types, value ranges, sensitive field verification)
- [ ] Extend each test with type assertions for numeric fields
- [ ] Add sensitive field verification (e.g., `api_key` absence check)
- [ ] Implement error-path test cases for invalid inputs
- [ ] Ensure error response structure is validated

**Effort estimation**
M

**Additional information**
Grooming confidence: High
Parent issue: #1097

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Closes #1145

The 5 affected integration tests only checked key presence. This PR adds explicit type assertions, value-range guards, a sensitive-field absence check, and error-path coverage to each ability.

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).

### What was tested

New test methods added (no data provider — self-contained):

| Ability | New methods |
|---|---|
| `GetStats` | `testStatsFieldsHaveCorrectTypes` — asserts `count_*` are `int ≥ 0`, sizes are `int ≥ 0`, `savings_percent` is `float` for both `wp` and `custom-folders` groups |
| `GetMediaStatus` | `testErrorResponseFieldTypesOnInvalidMediaId` — `media_id: 0` returns `status='error'`, correct field types; `testSuccessResponseFieldTypesForUnoptimizedAttachment` — factory attachment returns valid status string, int sizes, bool flags |
| `GetNextgenCoverage` | `testCoverageFieldsHaveCorrectTypes` — `missing_nextgen_count` is `int ≥ 0`, `nextgen_format` is `string` |
| `GetAccount` | `testApiKeyIsAbsentFromOutput` — verifies `api_key` key is never present; `testOutputFieldTypesMatchSchema` — all 7 schema fields are correct types |
| `OptimizeMedia` | `testErrorResponseFieldTypesOnInvalidMediaId` — `media_id: 0` returns null sizes and string message; `testErrorResponseForNonAttachmentPostId` — factory post (not attachment) is rejected with error status |

### How to test

Run the integration test suite against a WP 6.9+ environment:

```
cd Tests/Integration && php ../../vendor/bin/phpunit --filter "Abilities" --testdox
```

### Affected Features & Quality Assurance Scope

Test-only change. No production code modified.

### Documentation

All new test methods are self-contained (no data-provider fixture required). Key decisions:

- `GetMediaStatus` valid statuses are `success`, `error`, `unoptimized` (derived from `map_status()`).
- `GetAccount` type tests target the error-state path (no API key in integration environment), which still exercises the full type contract.
- `OptimizeMedia` non-attachment test exercises the post-type guard added in #1142.

### New dependencies

None.

### Risks

None. Test-only change.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

## Unticked items justification

N/A

# Additional Checks

- [ ] In the case of complex code, I wrote comments to explain it.
- [ ] When possible, I prepared ways to observe the implemented system (logs, data, etc.)
- [ ] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

Additional checks are unticked — test-only change with no complex logic and no I/O.

### CODE REVIEW

File: Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php [hunk 1]
```diff
@@ -49,6 +49,41 @@ public function testShouldReturnExpected( array $config, array $expected ): void
 		}
 	}
 
+	public function testErrorResponseFieldTypesOnInvalidMediaId(): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php:56]: There is no documentation here

File: Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php [hunk 2]
```diff
@@ -49,6 +49,41 @@ public function testShouldReturnExpected( array $config, array $expected ): void
...
+		$this->assertIsString( $result['error_message'] );
+		$this->assertNotEmpty( $result['error_message'] );
+		$this->assertNull( $result['optimization_level'] );
+		$this->assertIsInt( $result['original_size'] );
+		$this->assertIsInt( $result['optimized_size'] );
+		$this->assertIsBool( $result['webp_available'] );
+		$this->assertIsBool( $result['avif_available'] );
+	}
+
+	public function testSuccessResponseFieldTypesForUnoptimizedAttachment(): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php:76]: There is no documentation here

File: Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php [hunk 3]
```diff
@@ -49,6 +49,41 @@ public function testShouldReturnExpected( array $config, array $expected ): void
...
+		$this->assertContains( $result['status'], [ 'success', 'error', 'unoptimized' ] );
+		$this->assertIsInt( $result['original_size'] );
+		$this->assertIsInt( $result['optimized_size'] );
+		$this->assertGreaterThanOrEqual( 0, $result['original_size'] );
+		$this->assertGreaterThanOrEqual( 0, $result['optimized_size'] );
+		$this->assertIsBool( $result['webp_available'] );
+		$this->assertIsBool( $result['avif_available'] );
+	}
+
 	private function set_up_user( bool $has_permission ): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php:121]: There is no documentation here

File: Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php [hunk 4]
```diff
@@ -49,6 +49,18 @@ public function testShouldReturnExpected( array $config, array $expected ): void
 		}
 	}
 
+	public function testCoverageFieldsHaveCorrectTypes(): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php:56]: There is no documentation here

File: Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php [hunk 5]
```diff
@@ -49,6 +49,18 @@ public function testShouldReturnExpected( array $config, array $expected ): void
...
+
+		$ability = wp_get_ability( 'imagify/get-nextgen-coverage' );
+		$result  = $ability->execute();
+
+		$this->assertIsInt( $result['missing_nextgen_count'] );
+		$this->assertGreaterThanOrEqual( 0, $result['missing_nextgen_count'] );
+		$this->assertIsString( $result['nextgen_format'] );
+	}
+
 	private function set_up_user( bool $has_permission ): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php:95]: There is no documentation here

File: Tests/Integration/classes/Abilities/GetStats/RegisterAbilityTest.php [hunk 6]
```diff
@@ -49,6 +49,27 @@ public function testShouldReturnExpected( array $config, array $expected ): void
 		}
 	}
 
+	public function testStatsFieldsHaveCorrectTypes(): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/GetStats/RegisterAbilityTest.php:56]: There is no documentation here

File: Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php [hunk 7]
```diff
@@ -49,6 +49,37 @@ public function testShouldReturnExpected( array $config, array $expected ): void
 		}
 	}
 
+	public function testErrorResponseFieldTypesOnInvalidMediaId(): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php:56]: There is no documentation here

File: Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php [hunk 8]
```diff
@@ -49,6 +49,37 @@ public function testShouldReturnExpected( array $config, array $expected ): void
...
+
+		$this->assertSame( 'error', $result['status'] );
+		$this->assertNull( $result['original_size'] );
+		$this->assertNull( $result['optimized_size'] );
+		$this->assertNull( $result['savings_percent'] );
+		$this->assertIsString( $result['error_message'] );
+		$this->assertNotEmpty( $result['error_message'] );
+	}
+
+	public function testErrorResponseForNonAttachmentPostId(): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php:74]: There is no documentation here

File: Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php [hunk 9]
```diff
@@ -49,6 +49,37 @@ public function testShouldReturnExpected( array $config, array $expected ): void
...
+
+		$this->assertSame( 'error', $result['status'] );
+		$this->assertIsString( $result['error_message'] );
+		$this->assertNotEmpty( $result['error_message'] );
+		$this->assertNull( $result['original_size'] );
+		$this->assertNull( $result['optimized_size'] );
+		$this->assertNull( $result['savings_percent'] );
+	}
+
 	private function set_up_user( bool $has_permission ): void {
```
Comment (Honemo) [inline Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php:117]: There is no documentation here

**Review by Honemo — CHANGES_REQUESTED** [review]

File: Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php (hunk 1 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php:56]: Done

File: Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php (hunk 2 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php:76]: Done

File: Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php (hunk 3 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/GetMediaStatus/RegisterAbilityTest.php:121]: Done

File: Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php (hunk 4 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php:56]: Done

File: Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php (hunk 5 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/GetNextgenCoverage/RegisterAbilityTest.php:95]: Done

File: Tests/Integration/classes/Abilities/GetStats/RegisterAbilityTest.php (hunk 6 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/GetStats/RegisterAbilityTest.php:56]: Done

File: Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php (hunk 7 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php:56]: Done

File: Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php (hunk 8 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php:74]: Done

File: Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php (hunk 9 above)
Comment (Miraeld) [reply Tests/Integration/classes/Abilities/OptimizeMedia/RegisterAbilityTest.php:117]: Done

**Review by Honemo — APPROVED** [review]
